as/cNFT/assembly/market.ts

```diff
@@ -218,16 +219,24 @@ export function accept_bid(tokenId: string, bidder: string): void {
     promiseCreator.transfer(payout.get(token.creator_id))
 
     // Transfer bid share to previous owner
-    const promisePrevOwner = ContractPromiseBatch.create(token.prev_owner_id)
-    promisePrevOwner.transfer(payout.get(token.prev_owner_id))
+    if(token.prev_owner_id){
+        const promisePrevOwner = ContractPromiseBatch.create(token.prev_owner_id)
+        promisePrevOwner.transfer(payout.get(token.prev_owner_id))
+        env.promise_return(promisePrevOwner.id)
+    }
 
     env.promise_return(promiseCreator.id)
     env.promise_return(promiseOwner.id)
-    env.promise_return(promisePrevOwner.id)
 
     // Transfer token to bidder
-    const transferArgs: NftTransferArgs = { "token_id": tokenId, "bidder_id": bidder }
-    const promiseTransfer = ContractPromise.create(context.contractName, "nft_transfer", transferArgs, XCC_GAS, context.attachedDeposit)
+    const transferArgs: NftTransferArgs = { token_id: tokenId, receiver_id: bidder }
+    const promiseTransfer = ContractPromise.create(
+        context.contractName,
+        "nft_transfer",
+        transferArgs,
+        228000000000000,
```

Comment:
instead of hardcoding here save this value in a `const XCC_NFT_TRANSFER_GAS = 228000000000000`. Makes it clearer. Also this value might be a bit bigger that what you might have calculated at times, being all transactions are not taking exactly the same compute time so make sure to leave some room to account for this.

Reply:
Yeah. true 👍